course5/homework4/homework4.cc

```diff
@@ -44,7 +48,10 @@ class Stream {
         ~Stream()
         {
                 if (filename != nullptr) {
-                        std::fclose(fp);
+                        auto result = std::fclose(fp);
+                        if (result != 0) {
+                                throw std::runtime_error("Failed to properly close file");
```

Comment:
Do not never under any circumstances throw from destructor. Just ignore the error; you tried to close the file, you did your best, there is nothing more you can do.